modules/dbnd-airflow/src/dbnd_airflow/scheduler/single_dag_run_job.py

```diff
@@ -29,6 +30,36 @@
 logger = logging.getLogger(__name__)
 
 SCHEDUALED_OR_RUNNABLE = RUNNABLE_STATES.union({State.SCHEDULED})
+END_STATES = (State.SUCCESS, State.FAILED, State.UPSTREAM_FAILED)
+
+
+def _kill_zombies(dag_run, session):
+    """
+    Remove zombies DagRuns and TaskInstances.
+    If the job ended but the DagRun is still not finished then
+    fail the DR and related unfinished TIs
+    """
+    logger.info(
+        "Job has ended but the related %s is still in state %s - marking it as failed",
+        dag_run,
+        dag_run.state,
+    )
```

Comment:
This will render to something like
```
Job has ended but the related <DagRun test_dag @ 2020-09-12: backfill__something,  externally triggered False>
```